server/bleep/src/webserver/query.rs

```diff
@@ -625,6 +620,41 @@ impl ExecuteQuery for OpenReader {
             })
             .collect::<SmallVec<[_; 2]>>();
 
+        let top_docs = TopDocs::with_limit(50000);
+        let empty_collector = MultiCollector::new();
+
+        let relative_paths = open_directives
+            .iter()
+            .map(|d| d.relative_path.to_owned())
+            .collect::<Vec<_>>();
+
+        let collector = BytesFilterCollector::new(
+            indexer.source.raw_relative_path,
+            move |b| {
+                let Ok(relative_path) = std::str::from_utf8(b) else {
+                    return false;
+                };
+
+                // Check if *any* of the relative paths match. We can't compare repositories here
+                // because the `BytesFilterCollector` operates on one field. So we sort through this
+                // later. It's unlikely that a search will use more than one open query.
```

Comment:
this is achievable by chaining multiple filter collectors: `ByteFilterCollector::new(field1, filter1, BytesFilterCollector(field2, filter2, empty_collector))`